Respect LDFLAGS when building C files. Bug #435048
Patch hasn't been sent upstream yet.
Index: shotwell-0.23.2/Makefile
===================================================================
--- shotwell-0.23.2.orig/Makefile
+++ shotwell-0.23.2/Makefile
@@ -697,7 +697,7 @@ $(EXPANDED_C_FILES): $(VALA_STAMP)
 	@
 
 $(EXPANDED_OBJ_FILES): %.o: %.c $(CONFIG_IN) Makefile
-	$(CC) -c $(VALA_CFLAGS) $(CFLAGS) -o $@ $<
+	$(CC) -c $(VALA_CFLAGS) $(CFLAGS) $(LDFLAGS) -o $@ $<
 
 $(PROGRAM): $(EXPANDED_OBJ_FILES) $(RESOURCES) $(LANG_STAMP) $(THUMBNAILER_BIN) misc/gschemas.compiled $(DOC_LANG_STAMP)
 	$(CC) $(EXPANDED_OBJ_FILES) $(CFLAGS) $(LDFLAGS) $(RESOURCES) $(VALA_LDFLAGS) $(EXPORT_FLAGS) -o $@
Index: shotwell-0.23.2/plugins/Makefile.plugin.mk
===================================================================
--- shotwell-0.23.2.orig/plugins/Makefile.plugin.mk
+++ shotwell-0.23.2/plugins/Makefile.plugin.mk
@@ -54,7 +54,7 @@ $(CFILES): .stamp
 	@
 
 .c.o:
-	$(CC) -c $(CFLAGS) $(DEFINES) -I../.. -I ../common $<
+	$(CC) -c $(CFLAGS) $(LDFLAGS) $(DEFINES) -I../.. -I ../common $<
 
 $(PLUGIN).so: $(OFILES)
 	$(CC) $(LDFLAGS) -shared $(OFILES) $(LIBS) -o $@
